homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,429 @@
+        "Current net Power",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:solar-power",
+    ],
+    "net_electricity_cumulative": [
+        "Cumulative net Power",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+}
+
+MISC_SENSOR_MAP = {
+    "battery": ATTR_BATTERY_LEVEL,
+    "illuminance": ATTR_ILLUMINANCE,
+    "modulation_level": [
+        "Heater Modulation Level",
+        UNIT_PERCENTAGE,
+        "modulation",
+        "mdi:percent",
+    ],
+    "valve_position": ["Valve Position", UNIT_PERCENTAGE, None, "mdi:valve"],
+    "water_pressure": ATTR_PRESSURE,
+}
+
+INDICATE_ACTIVE_LOCAL_DEVICE = [
+    "cooling_state",
+    "flame_state",
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile sensors from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    all_devices = api.get_all_devices()
+    single_thermostat = api.single_master_thermostat()
+    for dev_id, entity in all_devices.items():
+        data = api.get_device_data(dev_id)
+        for sensor, sensor_type in {
+            **TEMP_SENSOR_MAP,
+            **ENERGY_SENSOR_MAP,
+            **MISC_SENSOR_MAP,
+        }.items():
+            if sensor in data:
+                if data[sensor] is None:
+                    continue
+
+                if "power" in entity["types"]:
+                    model = None
+
+                    if "plug" in entity["types"]:
+                        model = "Metered Switch"
+
+                    entities.append(
+                        PwPowerSensor(
+                            api,
+                            coordinator,
+                            entity["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                            model,
+                        )
+                    )
+                else:
+                    entities.append(
+                        PwThermostatSensor(
+                            api,
+                            coordinator,
+                            entity["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                        )
+                    )
+                _LOGGER.info("Added sensor.%s", entity["name"])
+
+        if single_thermostat is False:
+            for state in INDICATE_ACTIVE_LOCAL_DEVICE:
+                if state in data:
+                    entities.append(
+                        PwAuxDeviceSensor(
+                            api, coordinator, entity["name"], dev_id, DEVICE_STATE,
+                        )
+                    )
+                    _LOGGER.info(
+                        "Added auxiliary sensor.%s_state", "{}".format(entity["name"])
```

Comment:
```suggestion
                        "Added auxiliary sensor.%s_state", entity["name"]
```

Reply:
Also the entity id isn't guaranteed if there are conflicts so its best to change this to say which name its adding

Reply:
Changed accordingly